### ISSUE TITLE
rvierdiiev - MainVault.withdrawAllowance doesn't have ability to send funds to recipient

### ISSUE THREAD

**Issue by sherlock-admin** [sherlock-audit/2023-01-derby-judging#56]
rvierdiiev

medium

# MainVault.withdrawAllowance doesn't have ability to send funds to recipient

## Summary
`MainVault.withdrawAllowance` doesn't have ability to send funds to recipient. In case if msg.sender was blocked by `vaultCurrency` token(usdc, usdt), then it doesn't have ability to withdraw funds.
## Vulnerability Detail
MainVault.withdrawAllowance function allows caller to withdraw requested amount of `vaultCurrency` token. This token can be usdc or usdt which have ability to block users. Once user is blocked, he can't participate in token transfers.

In case if user was blocked before he wants to withdraw using `withdrawAllowance`, then function will revert as transfer will not be allowed. As result user will not be able to withdraw his funds.
This is because `MainVault.withdrawAllowance` doesn't have `_receiver` param. Note, that `withdraw` function [has this param](https://github.com/sherlock-audit/2023-01-derby/blob/main/derby-yield-optimiser/contracts/MainVault.sol#L133)
## Impact
Blocked user can't receive funds using `MainVault.withdrawAllowance` function.
## Code Snippet
https://github.com/sherlock-audit/2023-01-derby/blob/main/derby-yield-optimiser/contracts/MainVault.sol#L166-L179
## Tool used

Manual Review

## Recommendation
Add `_receiver` param to `MainVault.withdrawAllowance` function.

Duplicate of #240
